src/common/transformations/tests/common_optimizations/simplify_second_input_of_reshape_test.cpp

```diff
@@ -611,3 +611,53 @@ TEST_F(TransformationTestsF, SimplifySecondInputOfReshapeTest21) {
     }
     comparator.enable(FunctionsComparator::CONST_VALUES);
 }
+
+TEST_F(TransformationTestsF, SimplifySecondInputOfReshapeTestFalseSpecialZero) {
+    PartialShape data_shape{1, 128, 12, 64};
+    {
+        auto data = std::make_shared<opset7::Parameter>(element::f32, data_shape);
+
+        auto shape_of = std::make_shared<opset7::ShapeOf>(data);
+        auto gather_op = gather(shape_of, std::vector<int64_t>{0, 1});
+        auto constant = opset7::Constant::create(element::i64, Shape{1}, {768});
+        auto concat = std::make_shared<opset7::Concat>(OutputVector{gather_op, constant}, -1);
+
+        auto reshape = std::make_shared<opset7::Reshape>(data, concat, false);
+        model = std::make_shared<Model>(NodeVector{reshape}, ParameterVector{data});
+
+        manager.register_pass<ov::pass::SimplifySecondInputOfReshape>();
+    }
+    {
+        auto data = std::make_shared<opset7::Parameter>(element::f32, data_shape);
+        auto reshape_pattern = opset7::Constant::create(element::i64, Shape{3}, {0, 0, 768});
+        auto reshape = std::make_shared<opset7::Reshape>(data, reshape_pattern, true);
+        model_ref = std::make_shared<Model>(NodeVector{reshape}, ParameterVector{data});
+    }
+    comparator.enable(FunctionsComparator::ATTRIBUTES);
+    comparator.enable(FunctionsComparator::CONST_VALUES);
+}
+
+TEST_F(TransformationTestsF, SimplifySecondInputOfReshapeTestFalseSpecialZeroZeroDim) {
+    PartialShape data_shape{1, 0, 12, 64};
+    {
+        auto data = std::make_shared<opset7::Parameter>(element::f32, data_shape);
+
+        auto shape_of = std::make_shared<opset7::ShapeOf>(data);
+        auto gather_op = gather(shape_of, std::vector<int64_t>{0, 1});
+        auto constant = opset7::Constant::create(element::i64, Shape{1}, {768});
+        auto concat = std::make_shared<opset7::Concat>(OutputVector{gather_op, constant}, -1);
+
+        auto reshape = std::make_shared<opset7::Reshape>(data, concat, false);
+        model = std::make_shared<Model>(NodeVector{reshape}, ParameterVector{data});
+
+        manager.register_pass<ov::pass::SimplifySecondInputOfReshape>();
+    }
+    {
+        auto data = std::make_shared<opset7::Parameter>(element::f32, data_shape);
+        auto reshape_pattern = opset7::Constant::create(element::i64, Shape{3}, {0, 0, 768});
+        auto reshape = std::make_shared<opset7::Reshape>(data, reshape_pattern, true);
+        model_ref = std::make_shared<Model>(NodeVector{reshape}, ParameterVector{data});
+    }
+    comparator.enable(FunctionsComparator::ATTRIBUTES);
+    comparator.enable(FunctionsComparator::CONST_VALUES);
+}
```

Comment:
Can we use existing test and specialize them with `special_zero=false`? What's so special for these two cases.

Reply:
I believe we can improve this point in the next PR